internal/character/sushang/skill.go

```diff
@@ -0,0 +1,101 @@
+package sushang
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+var skillHits = []float64{0.3, 0.7}
+
+func (c *char) Skill(target key.TargetID, state info.ActionState) {
+	// check if target is broken
+	isBroken := false
+	if c.engine.Stats(target).Stance() == 0 {
+		isBroken = true
+	}
+
+	// 2 hits
+	for _, hitRatio := range skillHits {
+		c.engine.Attack(info.Attack{
+			Source:     c.id,
+			Targets:    []key.TargetID{target},
+			DamageType: model.DamageType_PHYSICAL,
+			AttackType: model.AttackType_SKILL,
+			BaseDamage: info.DamageMap{
+				model.DamageFormula_BY_ATK: skill[c.info.SkillLevelIndex()],
+			},
+			StanceDamage: 60.0,
+			EnergyGain:   30.0,
+			HitRatio:     hitRatio,
+		})
+	}
+
+	// sword stance hits
+	chances := [3]bool{}
+
+	if c.engine.HasModifier(c.id, UltBuff) {
+		for i := 1; i <= 2; i++ {
+			if isBroken || c.engine.Rand().Float64() < 0.33 {
+				chances[i] = true
+			}
+		}
+	}
+
+	if isBroken || c.engine.Rand().Float64() < 0.33 {
+		chances[0] = true
+	}
```

Comment:
Looking at dm, the non-ult hit should go **last** (does both ult hits, then full hit)